File: ProductManagement/ProductAPI/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProductAPI.Models
{
    public class Product
    {
        [Key]
        public int ProductId { get; set; }
        public string ProductCode { get; set; }
        [Required]
        public string Name { get; set; }
        public int Quantity { get; set; }
        [Required]
        public decimal Price { get; set; }
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }
        public int SubCategoryId { get; set; }
        public int CategoryId { get; set; }
    }
}

File: UI/src/model/product.ts
export {};
export interface Product {
    productId: number;
    productCode: string;
    productName: string;
    quantity: number;
    price: number;
    productDescription: string;
    imageUrl: string;
    categoryId : number;
    subCategoryId : number;
    categoryName : string;
    subcategoryName : string;
    image: File | null;
  }

